homeassistant/components/sungrow/config_flow.py

```diff
@@ -0,0 +1,83 @@
+"""Config flow for Sungrow Solar Energy integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from SungrowModbusClient import Client, sungrow_register
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("host"): str,
+        vol.Optional("port"): int,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    port = data.get("port", 502)
+    client = Client(sungrow_register.regmap, data["host"], port)
+
+    client.load_register()
+    if "4990 ~ 4999 - Serial number" not in client.inverter:
+        raise InvalidAuth
```

Comment:
if no auth is needed I can freely access all values. I picked one at random to determine if access was successful.

Reply:
Yes, better to pick an exception that fits the actual problem.  It's important because some of the exceptions turn into error messages that users will see on the UI